Realm/ObjectServerTests/RealmServer.swift

```diff
@@ -701,43 +712,83 @@ public class RealmServer: NSObject {
+        case .pbs(let bsonType):
+            appService = [
+                "name": "mongodb1",
+                "type": "mongodb",
+                "config": [
+                    "uri": "mongodb://localhost:26000",
+                    "sync": [
+                        "state": "enabled",
+                        "database_name": "test_data",
+                        "partition": [
+                            "key": "realm_id",
+                            "type": "\(bsonType)",
+                            "required": false,
+                            "permissions": [
+                                "read": true,
+                                "write": true
+                            ]
+                        ]
+                    ]
+                ]
+                ]
+        case .flx(let fields):
+            appService = [
+                "name": "mongodb1",
+                "type": "mongodb",
+                "config": [
+                    "uri": "mongodb://localhost:26000",
+                    "flexible_sync": [
+                        "state": "enabled",
+                        "database_name": "test_data",
+                        "queryable_fields_names": fields,
                         "permissions": [
-                            "read": true,
-                            "write": true
+                            "rules": [:],
+                            "defaultRoles": [[
+                                "name": "all",
+                                "applyWhen": [:],
+                                "read": true,
+                                "write": true
+                            ]]
                         ]
                     ]
                 ]
             ]
-        ])
+        }
 
+        let serviceResponse = app.services.post(appService)
         guard let serviceId = (try serviceResponse.get() as? [String: Any])?["_id"] as? String else {
             throw URLError(.badServerResponse)
         }
 
         let rules = app.services[serviceId].rules
 
-        // Creating the rules is a two-step process where we first add all the
-        // rules and then add properties to them so that we can add relationships
-        let schema = ObjectiveCSupport.convert(object: RLMSchema.shared())
         let syncTypes = schema.objectSchema.filter {
             guard let pk = $0.primaryKeyProperty else { return false }
             return pk.name == "_id"
         }
+
+        let partitionKeyType: String?
+        if case .pbs(let bsonType) = syncMode {
+            partitionKeyType = bsonType
+        } else {
+            partitionKeyType = nil
+        }
         var ruleCreations = [Result<Any?, Error>]()
         for objectSchema in syncTypes {
-            ruleCreations.append(rules.post(objectSchema.stitchRule(bsonType, schema)))
+            let validSyncClasses = ["Dog", "Person", "SwiftPerson", "SwiftTypesSyncObject"]
+            if partitionKeyType != nil ||
```

Comment:
This will be possible if all the added objects were Swift Objects, but because is not the case I was not able to add this.I simplified the code as much as possible so we don't have to change this twice.